### PR TITLE
[MAIN-168] Fix profile bug where games may be null due to being deleted

### PR DESCRIPTION
**Pull request by kevo1ution**
**Here is a demo of the fix**
bug_profile_not_viewable_fix_20220913

### CODE REVIEW

File: src/gameView/gamePlayer/RoomPlayer.tsx [hunk 1]
```diff
@@ -61,21 +58,7 @@ const RoomPlayer = () => {
     //          open user profiles)
     //       3. we should remove the original navbar, it feels unnecessary
     //       4. escape keybinding, but also physical button
-    return (
```
Comment (kevo1ution) [inline src/gameView/gamePlayer/RoomPlayer.tsx:64]: This is unecessary code, we can remove this modal

File: src/gameView/gamePlayer/IFrame/IFrame.tsx [hunk 2]
```diff
@@ -39,6 +40,19 @@ const IFrame = ({
...
 }: Props) => {
+  if (!room.game) {
+    return (
+      <Typography
+        marginTop="10px"
+        variant="h4"
+        align="center"
+        color="text.primary"
+      >
+        Game was deleted for the room!
```
Comment (sarahannali) [inline src/gameView/gamePlayer/IFrame/IFrame.tsx:51]: Maybe just "Game was deleted!" ? also if there is no room.game, does that automatically mean it was deleted? Should we just say 'Game not found'

File: src/profileView/ProfileView.tsx [hunk 3]
```diff
@@ -113,17 +115,35 @@ const ProfileView = ({ user, setUser }: Props) => {
...
                   // - disconnect happens when no session a player is connected is in the game
                   // (e.g. multiple tabs should only trigger it once)
-                  history.push(`/games/${room.game.id}/room/${room.id}`);
+
+                  // when game is hard deleted, we show a snackbar error because players can't play
+                  // a game that has been deleted
+                  if (room.game) {
+                    history.push(`/games/${room.game.id}/room/${room.id}`);
+                  } else {
+                    enqueueSnackbar('The original game got deleted. You cannot view rooms of deleted games.', {
```
Comment (sarahannali) [inline src/profileView/ProfileView.tsx:124]: Just 'This game was deleted.'

File: src/profileView/ProfileView.tsx [hunk 4]
```diff
@@ -113,17 +115,35 @@ const ProfileView = ({ user, setUser }: Props) => {
...
+                      variant: 'error',
+                      autoHideDuration: 3000,
+                    });
+                  }
                 }}
                 >
                   <CardHeader
-                    title={room.game.name}
+                    title={room.game ? room.game.name : '[Deleted Game]'}
```
Comment (sarahannali) [inline src/profileView/ProfileView.tsx:132]: should we also disable clicking the card / quit if there is no game

Comment (kevo1ution) [reply src/profileView/ProfileView.tsx:132]: I think we should actually just hide this room card

Comment (kevo1ution) [reply src/profileView/ProfileView.tsx:132]: we have a separate jira tracking handling doing room operations on a game that doesn't exist https://boardgames.atlassian.net/browse/MAIN-169

**Review by sarahannali — APPROVED** [review]

**Pull request merged**